### PR TITLE
Update for swup 4

### PR DESCRIPTION
**Pull request by daun**
**Checks**

- [x] The PR is submitted to the `master` branch
- [x] The code was linted before pushing (`npm run lint`)
- [x] The documentation was updated as required

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -71,67 +70,61 @@ export default class PreloadPlugin extends Plugin {
...
 		// Bail early if the visit should be ignored by swup
-		if (this.shouldIgnoreVisit(linkEl.href, { el: linkEl })) return;
+		if (this.swup.shouldIgnoreVisit(el.href, { el })) return;
 
 		// Bail early if the link points to the current page
 		if (url === getCurrentUrl()) return;
 
 		// Bail early if the page is already in the cache
-		if (swup.cache.exists(url)) return;
+		if (swup.cache.has(url)) return;
```
[the remark this answers is not in the document]
Comment (daun) [reply src/index.js:127]: How would this prevent an error? And what's the error?

[the remark this answers is not in the document]
Comment (daun) [reply src/index.js]: I see now!

[the remark this answers is not in the document]
Comment (hirasso) [reply src/index.js:127]: `swup` isn't defined in this context without `this`, isn't it?

[the remark this answers is not in the document]
Comment (daun) [reply src/index.js]: Ideally, this would have been caught by a solid eslint setup already.

[the remark this answers is not in the document]
Comment (daun) [reply src/index.js]: Looks like me having the debug plugin installed for testing made this work  as window.swup is now defined.

[the remark this answers is not in the document]
Comment (daun) [reply src/index.js]: Fixed!

[the remark this answers is not in the document]
Comment (hirasso) [reply src/index.js]: Oh, I just saw that this bug was introduced here by removing `const swup = this.swup` at the beginning of the function.

**Review by hirasso — APPROVED** [review]

**Pull request merged**